src/Database/Expression/WindowExpression.php

```diff
@@ -83,22 +94,102 @@ public function order($fields)
      */
     public function range(?int $start, ?int $end = 0)
     {
-        return $this;
+        if (func_num_args() === 1) {
+            return $this->frame(self::RANGE, $start, self::PRECEDING);
+        } else {
```

Comment:
The `else` is not necessary since the `if` part returns from the method. Same for other methods below.